arch/arm/core/aarch32/cortex_m/reset.S

```diff
@@ -78,6 +78,25 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 
 #endif /* CONFIG_INIT_ARCH_HW_AT_BOOT */
 
+    /*
```

Comment:
Is it possible that you don't have CONFIG_INIT_ATCH_HW_AT_BOOT enabled?  Commit 2ce570b03f97d5947e42fff77fe9437979c01c23 implies that the stack should work already when calling z_platform_init.  I wonder if you should just move the clearing of the SPLIM registers out so that it runs on your configuration.  On reset the main stack pointer is already setup, so it should be available.

Reply:
Thanks, I will close this PR.